stbt-run

```diff
@@ -45,35 +45,50 @@ stbt.debug("Arguments:\n" + "\n".join([
     "%s: %s" % (k, v) for k, v in args.__dict__.items()]))
 
 
-def _save_screenshot(exception):
+def _save_screenshot(dut, exception, save_jpg, save_png):
     import cv2
 
-    jpg = (args.save_thumbnail == 'always' or
-           (args.save_thumbnail == 'on-failure' and exception))
-    png = (args.save_screenshot == 'always' or
-           (args.save_screenshot == 'on-failure' and exception))
-
-    if not jpg and not png:
+    if not save_jpg and not save_png:
         return
 
     screenshot = getattr(exception, "screenshot", None)
-    if screenshot is None and stbt._dut._display:  # pylint: disable=protected-access
-        screenshot = stbt._dut._display.last_used_frame  # pylint: disable=protected-access
+    if screenshot is None and dut._display:  # pylint: disable=protected-access
+        screenshot = dut._display.last_used_frame  # pylint: disable=protected-access
     if screenshot is None:
-        screenshot = stbt._dut.get_frame()  # pylint: disable=protected-access
+        screenshot = dut.get_frame()  # pylint: disable=protected-access
 
-    if png:
+    if save_png:
         cv2.imwrite("screenshot.png", screenshot)
         sys.stderr.write("Saved screenshot to 'screenshot.png'.\n")
 
-    if jpg:
+    if save_jpg:
         cv2.imwrite(
             'thumbnail.jpg',
             cv2.resize(screenshot, (
                 640, 640 * screenshot.shape[0] // screenshot.shape[1])),
             [cv2.IMWRITE_JPEG_QUALITY, 50])
 
 
+@contextmanager
+def video(args):
+    dut = _stbt.core.new_device_under_test_from_config(args)
+    with stbt._set_dut_singleton(dut), dut:  # pylint: disable=protected-access
```

Comment:
Presumably `_set_dut_singleton` is added in a subsequent commit (I'm reviewing these in the order that github presents them which is by commit date not by git dag order, if I recall correctly).